### PR TITLE
GUIFormSpecMenu: Fix multiline translation

### LINKED ISSUE #7450: label[] breaks multiline translations
##### Issue type
 - Bug report
 
##### Minetest version
```
minetest-0.5.0-dev-9a1d3584
```

##### Summary
`label[x,y;text]` element breaks `minetest.get_translation` when the latter has line breaks.

##### Steps to reproduce
Write a following mod:
```lua
local S = minetest.get_translator("test")

minetest.register_on_joinplayer(function(player)
	minetest.show_formspec(player:get_player_name(), "test", ([[
		size[4,4]
		label[0,0;%s]
		textarea[0,2;4,2;;;%s]
	]]):format(S"A@nB@nC", S"A@nB@nC"))
end)
```

#### Expected Behavior
Both strings are translated into the same one or give a single warning.

#### Actual Behavior
If translation for `A@nB@nC` is present, only textarea is translated. I get 2 warnings about this:
```
2018-06-15 10:24:31: WARNING[Main]: Translations: can't find translation for string "A" in textdomain "test"
2018-06-15 10:24:31: WARNING[Main]: Translations: can't find translation for string "A
2018-06-15 10:24:31: WARNING[Main]: B
2018-06-15 10:24:31: WARNING[Main]: C" in textdomain "test"
```
First one is because of Label, second is because of Textarea.

### PR DESCRIPTION
**Pull request by SmallJoker**
Supersedes #12471
Fixes #7450

## To do

This PR is Ready for Review.

## How to test

1. https://github.com/minetest/minetest/pull/12471#issuecomment-1179601442
2. ???
3. maybe `/test_formspec`

### CODE REVIEW

File: src/gui/guiFormSpecMenu.cpp [hunk 1]
```diff
@@ -1731,25 +1731,27 @@ void GUIFormSpecMenu::parseLabel(parserData* data, const std::string &element)
...
-		std::wstring wlabel_plain = unescape_enriched(wlabel_colors);
+	EnrichedString str(unescape_string(utf8_to_wide(parts[1])));
+	size_t str_pos = 0;
+
+	for (size_t i = 0; str_pos < str.size(); ++i) {
+		// Split per line
+		size_t str_nl = str.getString().find(L'\n', str_pos);
+		if (str_nl == std::wstring::npos)
+			str_nl = str.getString().size();
+		EnrichedString line(str.substr(str_pos, str_nl - str_pos));
```
Comment (sfan5) [inline src/gui/guiFormSpecMenu.cpp:1753]: this is our custom substr method, right?
could it directly return an EnrichedString?

Comment (SmallJoker) [reply src/gui/guiFormSpecMenu.cpp:1753]: It already returns an enriched string.

Comment (sfan5) [reply src/gui/guiFormSpecMenu.cpp:1753]: weird to invoke the constructor then, what about this?
```suggestion
		EnrichedString line = str.substr(str_pos, str_nl - str_pos);
```

Comment (SmallJoker) [reply src/gui/guiFormSpecMenu.cpp:1753]: Right. I tend to use constructors over potential constructor + assign operators.

**Comment by doxygen-spammer** [comment]
Confirming the fix of multiline translations with arguments.

Thank you!

**Pull request merged**